Jenkinsfile

```diff
@@ -18,315 +18,326 @@ pipeline {
 		lock resource: 'IDLLock'
 	}
 	stages {
-		stage('Build image') {
+		stage("Build image") {
 			steps {
 				script {
 					dockerImage = docker.build("${DOCKER_IMG}")
 				}
 			}
 		}
-		stage('Push image') {
+		stage("Push image") {
 			steps {
 				script {
 					dockerImage.push("${GIT_DESCRIPTION}")
 				}
 			}
 		}
-		stage('DeployTestStack') {
+		stage("DeployTestStack") {
 			when {
 				anyOf {
-					branch 'PR-*'
+					branch "PR-*"
 					expression { return params.forceEndToEnd }
 				}
 			}
+	        steps {
+	            script {
+                    withKubeConfig([credentialsId: "prod-kube-config"]) {
+                        sh('helm repo update && helm upgrade --install nres-pipe lco/nres-pipe ' +
+                                '--set nresPipeline.tag="${GIT_DESCRIPTION}" --namespace dev --wait --timeout=3600')
 
-			steps {
-				script {
-					withCredentials([
-						usernamePassword(
-							credentialsId: 'rabbit-mq',
-							usernameVariable: 'RABBITMQ_USER',
-							passwordVariable: 'RABBITMQ_PASSWORD')
-						]) {
-							sh('rancher -c ${RANCHERDEV_CREDS} rm --stop --type stack NRESPipelineTest || true')
-							sh('rancher -c ${RANCHERDEV_CREDS} up --stack NRESPipelineTest --force-upgrade --confirm-upgrade -d')
-					}
-					CONTAINER_ID = getContainerId('NRESPipelineTest-NRESPipelineTest-1')
-					CONTAINER_HOST = getContainerHostName('NRESPipelineTest-NRESPipelineTest-1')
-				}
-			}
+                        podName = sh(script: 'kubectl -n dev get po -l app.kubernetes.io/instance=nres-pipe ' +
+                                        '--sort-by=.status.startTime -o jsonpath="{.items[-1].metadata.name}"',
+                                     returnStdout: true).trim()
+
+                    }
+                 }
+		    }
 		}
-		stage('Test-Bias-Ingestion') {
+		stage("Test-Bias-Ingestion") {
 			when {
 				anyOf {
-					branch 'PR-*'
+					branch "PR-*"
 					expression { return params.forceEndToEnd }
 				}
 			}
 			steps {
 				script {
-					sshagent(credentials: ['jenkins-rancher-ssh']) {
-						executeOnRancher('pytest --durations=0 --junitxml=/nres/code/pytest-bias-ingestion.xml ' +
-								'-m bias_ingestion /nres/code/',
-						    CONTAINER_HOST, CONTAINER_ID, ARCHIVE_UID)
+                    withKubeConfig([credentialsId: "prod-kube-config"]) {
+						sh("kubectl -n dev exec -c nres-pipeline ${podName} -- " +
+						        "/usr/bin/sudo -s -u archive pytest --durations=0 " +
+						        "--junitxml=/nres/code/pytest-bias-ingestion.xml -m bias_ingestion /nres/code/")
 					}
 				}
 			}
 			post {
 				always {
-					script{
-						sshagent(credentials: ['jenkins-rancher-ssh']) {
-							copyFromRancherContainer('/nres/code/pytest-bias-ingestion.xml', '.', CONTAINER_HOST, CONTAINER_ID)
+					script {
+					    withKubeConfig([credentialsId: "prod-kube-config"]) {
```

Comment:
If all these steps use the `dev` namespace, you can use the `dev-kube-config` credentials.